src/PrestaShopBundle/DataCollector/HookRegistry.php

```diff
@@ -0,0 +1,161 @@
+<?php
+/**
+ * 2007-2017 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2017 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\DataCollector;
+
+use PrestaShop\PrestaShop\Core\Module\ModuleInterface;
+
+/**
+ * Store every hook dispatched during a request.
+ */
+class HookRegistry
+{
+    /**
+     * @var array the selected hook.
+     */
+    private $hook;
```

Comment:
I'm a little confused about `hook` and `hooks`. In a HookRegistry, I would expect to have a collection of called hooks, which would be objects of their own. To me, it looks like this class has mixed concerns.

Reply:
I wanted to create a class `Hook` to store theses items, but this become even more confusing as you may expect instance of `HookCore` when it's just collected hook data.

I don't have perfect solution here, but as this won't (this shouldn't?) be extended or reused I think it's ok: I'll put theses classes as `final`.

I can rename `$hook` to `$selectedHook` and `$hooks` to `$hooksData`, what do you think?

Reply:
> I wanted to create a class Hook to store theses items, but this become even more confusing as you may expect instance of HookCore when it's just collected hook data.

Problem solved: It's not a Hook, it's a `HookCall` or `HookCallMetadata` 😉 .

I think it would be better to separate the concerns, otherwise it's too confusing. You could also change this class name to `HookCallRegistry` to make it clearer if you want.

Regardless of if it will or not be extended or reused, that's no excuse for sloppy design. "Good enough for now" code is the basis of future "horror story" legacy code. Let's try and keep that to a minimum! 🙂